src/pages/Signup.tsx

```diff
@@ -0,0 +1,33 @@
+import React, { useState } from 'react';
+import TextInput from '../components/TextInput';
+import Button from '../components/Button';
+import { supabase } from '../App';
+
+const SignUp = () => {
+  const [email, setEmail] = useState<string>('');
+  const [password, setPassword] = useState<string>('');
+
+  const submitToSupabase = async () => {
+    const { data, error } = await supabase.auth.signUp({
```

Comment:
Here we have repetition with our interactions with supabase. We could centralise these in one action file and perhaps condense into one exported action that can handle multiple scenarios.